### PR TITLE
Add versionbump script and changelog file.

### PR DESCRIPTION
**Pull request by diaena**
See issue https://github.com/shibatch/sleef/issues/46

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Changelog
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
+and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## v[3.0.1] - 2017-07-19
```
Comment (fpetrogalli-arm) [inline CHANGELOG.md]: I believe this is 3.1.0

File: bumpversion.sh [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  [[ "$INPUT_ADDED" = "" ]] && break;
+  echo "- $INPUT_ADDED" >> tmpfile
+done
+
+# Add entry
+entry="$(cat tmpfile)" && rm tmpfile
+awk -ventry="$entry" '/^#.*Unreleased/{print;print entry;next}1' CHANGELOG.md > CHANGELOG.md.tmp
+mv CHANGELOG.md.tmp CHANGELOG.md
+
+# Push tags to git
```
Comment (fpetrogalli-arm) [inline bumpversion.sh]: I am not comfortable with this automatism. I would rather add comments saying what is the procedure for committing the new Changelog manually when ready.

File: bumpversion.sh [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+git add CHANGELOG.md VERSION.txt
+git commit -m "Version bump to $VERSION_BUMP"
+git tag -a -m "Tagging version $VERSION_BUMP" "v$VERSION_BUMP"
+git push origin --tags
```
Comment (fpetrogalli-arm) [inline bumpversion.sh]: Fine to have the commit commands automatic, not the push.

Please explain the "send to github" bits as comments.

File: CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+- DFT (Discrete Fourier Transform) functions
+- sincospi functions
+- Files needed for creating a debian package
+### Changed:
+- gencoef now supports single, extended and quad precision in addition to double precision
+- The library can be compiled as DLLs
+### Removed:
+### Fixed:
+
+## v[3.11.0] - 2017-81-30
```
Comment (fpetrogalli-arm) [inline CHANGELOG.md]: Wrong version and date

Comment (diaena) [inline CHANGELOG.md]: I have no idea how this got in.

Comment (fpetrogalli-arm) [inline CHANGELOG.md]: Was this auto-generated with the script? If so, there might be a bug in it.

Comment (diaena) [inline CHANGELOG.md]: Yes, but that does not explain the date :(

**Comment by shibatch** [comment]
Is the script  fully developed by Diana?
Please add copyright notice.

**Comment by diaena** [comment]
Are we okay to merge this?

**Comment by diaena** [comment]
Anything else that needs doing here?

**Comment by shibatch** [comment]
You can merge.

**Pull request closed without merging**